Doc segmentation algorithm (#367)
DassHydro/smash: doc/source/user_guide/classical_uses/hydrograph_segmentation.rst

Hydrograph Segmentation
=======================

> This section aims to go into detail on how to use and visualize the hydrograph segmentation from 
> a proposed algorithm as depicted in section :ref:`Math / Num Documentation <math_num_documentation.hydrograph_segmentation>`.

Reviewer: ```suggestion
This section provides a detailed guide on using and visualizing hydrograph segmentation
based on the algorithm depicted in section :ref:`Math / Num Documentation <math_num_documentation.hydrograph_segmentation>`.
```

First, open a Python interface:

.. code-block:: none

    python3

Imports
*******

> 

Reviewer: ```suggestion

We will first import everything we need in this tutorial.

```

.. ipython:: python
    
    import smash
    import matplotlib.pyplot as plt
    import pandas as pd

Model object creation
*********************

> To obtain flood events segmentation, you need to create a :class:`smash.Model` object. 
> For this case, we will use the ``Cance`` dataset used in the User Guide section: :ref:`user_guide.demo_data.cance`.

Reviewer: ```suggestion
Now, we need to create a :class:`smash.Model` object.
For this case, we will use the :ref:`user_guide.demo_data.cance` dataset as an example.
```
